pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -1026,8 +1026,30 @@ impl DeltaLayerInner {
                 if Some(meta.meta.key) == ignore_key_with_err {
                     continue;
                 }
+                let slice = &blobs_buf.buf[meta.start..meta.end];
+                let decompressed = vectored_blob_io::decompress(slice, meta.compression_bits).await;
+                let decompressed = match decompressed {
+                    Ok(buf) => buf,
+                    Err(e) => {
+                        reconstruct_state.on_key_error(
+                            meta.meta.key,
+                            PageReconstructError::Other(anyhow!(e).context(format!(
+                                "Failed to decompress blob from virtual file {}",
+                                self.file.path,
+                            ))),
+                        );
+
+                        ignore_key_with_err = Some(meta.meta.key);
+                        continue;
```

Comment:
Wondering if we can use a closure / fake `async move` block to dedupe the `if-err-then-call-on_key_error-and-set-ignore_key_with_err-and-continue` logic with what we have in line 1056.

Reply:
I tried to do this but there is no generics in closure. I came up with this using a function but not sure how much it deduped: https://github.com/neondatabase/neon/pull/9123

Reply:
ok, #9123 doesn't look convincing, bummer